php/example_code/auto-scaling/AutoScalingService.php

```diff
@@ -0,0 +1,193 @@
+            'AvailabilityZones' => $availabilityZones,
+            'MinSize' => $minSize,
+            'MaxSize' => $maxSize,
+            'LaunchTemplate' => [
+                'LaunchTemplateId' => $launchTemplateId,
+            ],
+        ]);
+    }
+    #snippet-end:[php.example_code.auto-scaling.service.createAutoScalingGroup]
+
+    #snippet-start:[php.example_code.auto-scaling.service.updateAutoScalingGroup]
+    public function updateAutoScalingGroup($autoScalingGroupName, $args)
+    {
+        if (array_key_exists('MaxSize', $args)) {
+            $maxSize = ['MaxSize' => $args['MaxSize']];
+        } else {
+            $maxSize = [];
+        }
+        if (array_key_exists('MinSize', $args)) {
+            $minSize = ['MinSize' => $args['MinSize']];
+        } else {
+            $minSize = [];
+        }
+        $parameters = ['AutoScalingGroupName' => $autoScalingGroupName];
+        $parameters = array_merge($parameters, $minSize, $maxSize);
+        return $this->autoScalingClient->updateAutoScalingGroup($parameters);
+    }
+    #snippet-end:[php.example_code.auto-scaling.service.updateAutoScalingGroup]
+
+    #snippet-start:[php.example_code.auto-scaling.service.deleteAutoScalingGroup]
+    public function deleteAutoScalingGroup($autoScalingGroupName)
+    {
+        return $this->autoScalingClient->deleteAutoScalingGroup([
+            'AutoScalingGroupName' => $autoScalingGroupName,
+            'ForceDelete' => true,
+        ]);
+    }
+    #snippet-end:[php.example_code.auto-scaling.service.deleteAutoScalingGroup]
+
+    #snippet-start:[php.example_code.auto-scaling.service.describeAutoScalingGroups]
+    public function describeAutoScalingGroups($autoScalingGroupNames)
+    {
+        return $this->autoScalingClient->describeAutoScalingGroups([
+            'AutoScalingGroupNames' => $autoScalingGroupNames
+        ]);
+    }
+    #snippet-end:[php.example_code.auto-scaling.service.describeAutoScalingGroups]
+
+    public function waitUntilGroupInService($autoScalingGroupNames)
+    {
+        $this->autoScalingClient->waitUntil('GroupInService', ['AutoScalingGroupNames' => $autoScalingGroupNames]);
+    }
+
+    #snippet-start:[php.example_code.auto-scaling.service.terminateInstanceInAutoScalingGroup]
+    public function terminateInstanceInAutoScalingGroup(
+        $instanceId,
+        $shouldDecrementDesiredCapacity = true,
+        $attempts = 0
+    ) {
+        try {
+            return $this->autoScalingClient->terminateInstanceInAutoScalingGroup([
+                'InstanceId' => $instanceId,
+                'ShouldDecrementDesiredCapacity' => $shouldDecrementDesiredCapacity,
+            ]);
+        } catch (AutoScalingException $exception) {
+            if ($exception->getAwsErrorCode() == "ScalingActivityInProgress" && $attempts < 5) {
+                error_log("Cannot terminate an instance while it is still pending. Waiting then trying again.");
+                sleep(5 + (1 * $attempts));
```

Comment:
Isn't this `5 * $attempts`? Or is there some weird coercion going on?

Reply:
Hm, I think I meant that to be `5 * (1 + $attempts)`. I'll fix it.

Reply:
I could also start $attempts at 1 instead of 0 if you think that's cleaner.

Reply:
Oh yeah you don't want to sleep for 0. `5 * (1 + $attempts)` probably best captures the intent.